okta/appUser.go

```diff
@@ -0,0 +1,101 @@
+/*
+* Copyright 2018 - Present Okta, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// AUTO-GENERATED!  DO NOT EDIT FILE DIRECTLY
+
+package okta
+
+import (
+	"time"
+	"github.com/okta/okta-sdk-golang/okta/query"
+	"fmt"
+)
+
+type AppUserResource resource
+
+type AppUser struct {
+	Embedded []string `json:"_embedded,omitempty"`
+	Links []string `json:"_links,omitempty"`
+	Created *time.Time `json:"created,omitempty"`
+	Credentials *AppUserCredentials `json:"credentials,omitempty"`
+	ExternalId string `json:"externalId,omitempty"`
+	Id string `json:"id,omitempty"`
+	LastSync *time.Time `json:"lastSync,omitempty"`
+	LastUpdated *time.Time `json:"lastUpdated,omitempty"`
+	PasswordChanged *time.Time `json:"passwordChanged,omitempty"`
+	Profile []string `json:"profile,omitempty"`
+	Scope string `json:"scope,omitempty"`
+	Status string `json:"status,omitempty"`
+	StatusChanged *time.Time `json:"statusChanged,omitempty"`
+	SyncState string `json:"syncState,omitempty"`
+}
+
+func (m *AppUser) WithCredentials(v *AppUserCredentials) *AppUser {
```

Comment:
In general if a field on a struct is exported, there isn't any need to add a setter. Unless there is some processing or validation you want to do when the fields are set, you should just let the user set them directly.

If you don't want users setting it directly, you shouldn't export it and provide getters/setters. This doesn't really apply in this case as you can't (un)marshal unexported fields.